### DISCUSSION TITLE
Issue with the API for using in picoharp 300

### DISCUSSION THREAD

**Discussion by yonghu92** [PicoQuant/snAPI#32] [Q&A] [answered]
Hi,  I am trying to use the snAPI for control the picoharp 300. However, it doesn't work for me. The error I saw is (attached image):
 
![image](https://github.com/user-attachments/assets/5de72cfd-1e02-478e-b4e0-13f3ef2c0494)
The code I am trying to run is the demo time trace code 
 
Could you please let me knwo if the API also works for picoharp 300? What kind of cif figure I should use for picoharp 300.
In additional, could you please let me know how to fix the issure with my code.

**Comment by tpoint75** [comment, marked as answer]
I'm sorry but PH300 is not supported!
https://picoquant.github.io/snAPI/introduction.html
"snAPI is free to use and compatible with PicoQuant [TimeHarp 260](https://picoquant.github.io/snAPI/introduction.html#https://www_picoquant_com/products/category/tcspc-and-time-tagging-modules/timeharp-260-tcspc-and-mcs-board-with-pcie-interface) (drivers optional),
[MultiHarp 160](https://picoquant.github.io/snAPI/introduction.html#https://www_picoquant_com/products/category/tcspc-and-time-tagging-modules/multiharp-160), [MultiHarp 150](https://picoquant.github.io/snAPI/introduction.html#https://www_picoquant_com/products/category/tcspc-and-time-tagging-modules/multiharp-150-high-throughput-multichannel-event-timer-tcspc-unit), [HydraHarp 400](https://picoquant.github.io/snAPI/introduction.html#https://www_picoquant_com/products/category/tcspc-and-time-tagging-modules/hydraharp-400-multichannel-picosecond-event-timer-tcspc-module) and [PicoHarp 330](https://picoquant.github.io/snAPI/introduction.html#https://www_picoquant_com/products/category/tcspc-and-time-tagging-modules/picoharp_330_precise_and_versatile_event_timer_and_tcspc_unit)."
